### DISCUSSION TITLE
Forms on public facing pages styling issues

### DISCUSSION THREAD

**Discussion by thairish** [filamentphp/filament#6895] [Help]
Hi,

I am building a Laravel Livewire project that uses Filament for the admin panel and some front end forms. I have set up the admin panel from the installation instructions: https://filamentphp.com/docs/2.x/admin/installation

This is working.

The issue I'm facing is trying to setup Filament forms on a public facing page. The validation doesn't look to be working and the styling for the DatePicker and FileUpload fields are not working.

My Livewire class implements HasForm:

```
class RegisterForm extends Component implements HasForms
```

The class includes the relevant imports:

```
use Carbon\Carbon;
use Filament\Forms;
use Livewire\Component;
use Filament\Forms\Contracts\HasForms;
use Filament\Forms\Components\DatePicker;
use Filament\Forms\Components\FileUpload;
use Filament\Forms\Components\TextInput;
```

This is the schema function:

```
protected function getFormSchema(): array 
    {
        return [
            TextInput::make('first_name')
                ->required(),
            TextInput::make('middle_name'),
            TextInput::make('last_name')
                ->required(),
            TextInput::make('email')
                ->rules([
                    function () {
                        return function (string $attribute, $value, Closure $fail) {
                            Log::info('logging to see if this is called');

                            $fail("Force a validation error");
                        };
                    },
                ])
                ->required(),

            TextInput::make('password')
                ->password()
                ->same('password_confirmation')
                ->required(),
            TextInput::make('password_confirmation')
                ->password()
                ->required(),
            FileUpload::make('document')
                ->label('document')
                ->disk('s3')
                ->directory('user-documents')
                ->visibility('private')
                ->acceptedFileTypes(['application/pdf'])
                ->maxSize(1024)
                ->required(),
            DatePicker::make('expiry_date')
                ->label('Document expiry date')
                ->minDate(Carbon::now())
                ->required(),
        ];
    } 
```

The "submit" function does get called when the submit button is clicked.

All of the fields do display, however, both the date picker and file upload haven't got the styling pulling through. Apart from the file upload PDF validation message displaying it doesn't look like any other validation is working as I don't see any validation errors on submit (and the log on the email custom rule isn't called).

There are no errors in my laravel.log file and no JS errors in Chrome tools.

I've tried installation steps from the documentation: https://filamentphp.com/docs/2.x/forms/installation

All of the steps were fine but I am having issues with the "Configuring styles" section. When I add the CSS line:

```
@import '../../vendor/filament/forms/dist/module.esm.css';
```

I receive the error:

```
Internal server error: [postcss] C:\project-path\vendor\filament\forms\dist\module.esm.css:1618:3: The `bg-primary-600` class does not exist. If `bg-primary-600` is a custom class, make sure it is defined within a `@layer` directive.
```

This is not a custom CSS class that I've setup in Tailwind. I'm assuming this is what's causing the styling to be messed up for the date picker and file upload, but I'm unsure why the validation isn't working.

Please can anyone advise a solution or steps for further debugging?

Thanks

**Comment by thairish** [comment]
I've managed to resolve the validation issue, forgot to add $this->validate() in the submit() function. However, the styling issue still persists.

**Comment by danharrin** [comment]
Sounds like you havent configured the tailwind.config.js as per the form builder installation instructions, otherwise `module.esm.css` would work.

**Reply by thairish** [reply]
Thanks for your reply. I've double checked and I was missing:

```
'./vendor/filament/**/*.blade.php',
```

This cleared up the DatePicker styling but unfortunately I'm still getting that postcss error. The FileUpload styling is still missing even though the functionality works.

I've also tried a complete copy and paste of the tailwind.config.js from the documentation but no luck.

**Reply by danharrin** [reply]
You are missing the colors. Maybe you didn't import them.
